packages/react-codemirror/src/lang-cypher/syntax-validation.ts

```diff
@@ -55,7 +55,13 @@ export const semanticAnalysisLinter: (config: CypherConfig) => Extension = (
 
     // we want to avoid the ANTLR4 reparse in the worker thread, this should hit our main thread cache
     const parse = parserWrapper.parse(query);
-    if (parse.diagnostics.length !== 0) {
+    const statements = parse.statementsParsing;
+
+    const anySyntacticError =
+      statements.filter((statement) => statement.diagnostics.length !== 0)
+        .length > 0;
```

Comment:
For convenience it could be interesting to have something aggregating helper functions on the top level parse result?  
```ts
const parse = parserWrapper.parse(query);
if(parse.getAllDiagnostics().length > 0) {
```

Reply:
Wouldn't that be creating extra objects (cause we need to concatenate possibly a few arrays from the internal statements)? 

Not sure there's a logical view for arrays in Typescript or a similar concept